src/auth/AppAuth0ClientPromise.tsx

```diff
@@ -0,0 +1,36 @@
+import createAuth0Client from "@auth0/auth0-spa-js";
+import Auth0Client from "@auth0/auth0-spa-js/dist/typings/Auth0Client";
+
+import staticAuthConfig from "./auth_config.json";
+
+let redirectUri =
+  typeof window === "undefined" ? undefined : window.location.origin;
+
+const finalAuthConfig = {
+  domain: staticAuthConfig.domain,
+  // eslint-disable-next-line @typescript-eslint/camelcase
+  client_id: staticAuthConfig.clientId,
+  audience: staticAuthConfig.audience,
+  // eslint-disable-next-line @typescript-eslint/camelcase
+  redirect_uri: redirectUri,
+};
+
+/**
+ * A Promise that resolves to an Auth0 client for this app. This is intended to
+ * be used as a global variable -- all users of this variable will see the same
+ * instance, which makes sure that we only make one network request to check the
+ * auth status.
+ *
+ * Creating the Auth0 client was originally done in
+ * src/auth/react-auth0-spa.tsx, but the non-React-aware code in
+ * src/database/index.tsx needs access to this client as well.
+ *
+ * TODO: Refactor the database code to accept an Auth0 client, or something like
+ * that.
+ */
+const AppAuth0ClientPromise: Promise<Auth0Client> =
+  typeof window === "undefined"
+    ? new Promise(() => undefined)
```

Comment:
What are the cases in which we may reach this undefined state, and what is the impact on app behavior?

Reply:
All the times when window is `undefined` are during static render. When the code is loaded by the browser, window is always defined! 

This check is done a few times in the app. Let me know what a good way to document this is. I thought about abstracting it into a function like `isStaticRender`, but I'm not sure if such a function would make it less clear why we need the check. I can also just add a short comment (`// window is undefined during static rendering`) every time we have this check.

Reply:
Yeah I think another layer of indirection on that would probably harm legibility, I agree. Just that short comment would probably suffice, but we may not even need it. I was more curious than anything.

Reply:
Fixed in 6b1bf93